src/java/htsjdk/samtools/SamPairUtil.java

```diff
@@ -466,6 +467,7 @@ private void advance() {
         }
 
         public SAMRecord next() {
+            if(!hasNext()) { throw new NoSuchElementException(); }

```

Comment:
Sorry @alecw I wasn't clear enough. I meant the assertOpen() in htsjdk.samtools.SAMRecord#hasNext() and htsjdk.samtools.SAMRecord#next()

shouldn't hasNext() simply return false if the iterator is closed, not throw an exception? And the exception being thrown in this case should be a NoSuchElementException rather than a AssertionError, no?  The only reason I mentioned you here was that you seemed to have added the AbstractBamIterator about two years ago, and together with that change, added the calls to assertOpen.